### PR TITLE
fixed #17594: [android] fix crash and memory leaks if launching game multiple times on Android platform with PhysX module enabled.

### LINKED ISSUE #17594: Crash if launching game multiple times on Android platform with PhysX module enabled.
Reported at https://forum.cocos.org/t/topic/161024

### PR DESCRIPTION
**Pull request by dumganhar**
Re: #17594

https://github.com/cocos/google-game-sdk/pull/9
https://github.com/cocos/cocos-engine-external/pull/458

### Changelog

*

-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

## Greptile Summary

This pull request updates the external dependency version for the engine-native-external repository, addressing crash and memory leak issues on Android platforms with PhysX module enabled.

- Updated `native/external-config.json` to use engine-native-external version v3.8.5-1, up from v3.8.4-5
- This change likely incorporates fixes for the ADPFManager class, including proper resource cleanup and improved header inclusion

### CODE REVIEW

File: native/cocos/scene/RenderWindow.cpp [hunk 1]
```diff
@@ -56,7 +56,17 @@ RenderWindow::RenderWindow()
   _colorName("Color" + std::to_string(_renderWindowId)),
   _depthStencilName("DepthStencil" + std::to_string(_renderWindowId)) {}
 
-RenderWindow::~RenderWindow() = default;
+RenderWindow::~RenderWindow() {
+    // NOTE: destroy needs to be invoked in the destructor of RenderWindow to avoid wild pointer issues in gfx backend code.
+    // RenderWindow owns `_frameBuffer` and `_colorTextures`, `_frameBuffer` should be released before `_colorTextures`
+    // since gfx::Framebuffer keeps a weak pointer of `_colorTextures` and there is code :
+    // GLES3Device::getInstance()->framebufferHub()->disengage(colorTexture->gpuTexture(), _gpuFBO);
```
Comment (minggo) [inline native/cocos/scene/RenderWindow.cpp:63]: Why not FrameBuffer check validation of colortexture before using it as it is just a weak pointer.

Comment (dumganhar) [reply native/cocos/scene/RenderWindow.cpp:63]: colorTexture is released in RenderWindow and Framebuffer owns a weak pointer which is a wild pointer, we could not check whether a pointer is wild.

Comment (minggo) [reply native/cocos/scene/RenderWindow.cpp:63]: Ok, i thought "weak pointer" is a type of smart pointer. It is a raw pointer.

**Review by minggo — APPROVED** [review]

**Comment by dumganhar** [comment]
@cocos-robot run test cases

**Pull request merged**